datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1035,15 +1039,70 @@ async fn unnest_columns() -> Result<()> {
     // Test aggregate results for points and tags.
     let df = table_with_nested_types(NUM_ROWS).await?;
     let count = df
-        .unnest_column("points")?
-        .unnest_column("tags")?
+        .unnest_column("points", preserve_nulls)?
+        .unnest_column("tags", preserve_nulls)?
         .count()
         .await?;
     assert_eq!(count, results.iter().map(|r| r.num_rows()).sum::<usize>());
 
     Ok(())
 }
 
+#[tokio::test]
+async fn unnest_column_nulls() -> Result<()> {
```

Comment:
Here is a test that I think shows the difference between the two modes (or will once we implement it)